cl/lib/storage.py

```diff
@@ -1,45 +1,61 @@
 import itertools
 import os
 import uuid
+from typing import Optional
 
 from django.conf import settings
-from django.core.files.storage import FileSystemStorage
+from django.core.files.storage import FileSystemStorage, Storage
 from storages.backends.s3boto3 import S3Boto3Storage
 
 
-class IncrementingFileSystemStorage(FileSystemStorage):
-    """This maintains the old behavior of get_available_name that was available
+def get_name_by_incrementing(
+    instance: Storage,
+    name: str,
+    max_length: Optional[int] = None,
+) -> str:
+    """Generate usable file name for storage iterating if needed.
+
+    Returns a filename that is available in the storage mechanism,
+    taking the provided filename into account.
+
+    This maintains the old behavior of get_available_name that was available
     prior to Django 1.5.9. This behavior increments the file name by adding _1,
     _2, etc., but was removed because incrementing the file names in this
-    manner created a security vector if users were able to upload files.
+    manner created a security vector if users were able to upload (many) files.
 
     We are only able to use it in places where users are not uploading files,
-    and we are instead creating them programatically (for example, via a
+    and we are instead creating them programmatically (for example, via a
     scraper).
 
     For more detail, see:
 
     https://docs.djangoproject.com/en/1.8/releases/1.5.9/#file-upload-denial-of-service
+
+    :param instance: The instance of the storage class being used
```

Comment:
This is very minor, but I would either use punctuation on all params - or none

Reply:
Good call. I'll clean it up.